Celeste.Mod.mm/Mod/Registry/StrawberryRegistry.cs

```diff
@@ -0,0 +1,154 @@
+        {
+            ReadOnlyCollection<RegisteredBerry> berries = GetTrackableBerries();
+            foreach (RegisteredBerry berry in berries)
+            {
+                if (berry.entityName == name)
+                    return true;
+            }
+            return false;
+        }
+
+        // Is it the first normally collectable strawberry in the train?
+        public static bool IsFirstStrawberry(Entity self)
+        {
+            Follower follow = self.Components.Get<Follower>();
+            if (follow == null)
+                return false;
+
+            // we need it searchable and Find doesn't work on a ReadOnlyCollection aaaaa
+            List<RegisteredBerry> berries = registeredBerries.ToList();
+            ReadOnlyCollection<Type> types = GetBerryTypes();
+            //RegisteredBerry detectedBerry = berries.Find(b => b.GetType() == self.GetType());
+            //if (detectedBerry != null)
+            //if (detectedBerry == null)
+            //    return false;
+
+            for (int i = follow.FollowIndex - 1; i >= 0; i--)
+            {
+                Entity strawberry = follow.Leader.Followers[i].Entity;
+                bool goldenCheck = false;
+                if (strawberry as Strawberry != null)
+                    goldenCheck = (strawberry as Strawberry).Golden;
+
+                //Strawberry vanillaStraw = strawberry as Strawberry;
+                //if (vanillaStraw != null && !vanillaStraw.Golden)
+                //{
+                //    return false;
+                //}
+                Type t = follow.Entity.GetType();
+                RegisteredBerry berry = berries.Find(b => b.berryClass == t);
+                bool blocksCollect = berry.blocksNormalCollection;
+                if (types.Contains(strawberry.GetType()) && 
+                    berry.blocksNormalCollection == false && 
+                    !goldenCheck)
+                {
+                    return false;
+                }
+            }
+            return true;
+        }
+
+        // Called by CoreMapDataProcessor in order to inject mod berries into the step list for fixup.
+        public static Dictionary<string, Action<BinaryPacker.Element>> GetBerriesToInject(MapDataFixup context)
+        {
+            Dictionary<string, Action<BinaryPacker.Element>> trackDict = new Dictionary<string, Action<BinaryPacker.Element>>();
+            ReadOnlyCollection<RegisteredBerry> berries = GetRegisteredBerries();
+            foreach (RegisteredBerry berry in berries)
+            {
+                trackDict.Add(String.Concat("entity:", berry.entityName), entity => { context.Run("entity:strawberry", entity); });
+            }
+
+            return trackDict;
+        }
+
+        public class RegisteredBerry
+        {
+            // registered berry as a Type
+            public readonly Type berryClass;
+
+            // entity:<passed berry's defined name>
+            public readonly string entityName;
```

Comment:
As far as I can tell this string will never contain `entity:` at the beginning. Can you please clarify or fix the comment above it?

Reply:
Doodled on it with some crayon to make it better.